src/App.tsx

```diff
@@ -14,11 +15,81 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  nameGood: string,
+  selectedGood: string,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    nameGood: 'Jam',
+    selectedGood: 'Jam is selected',
+  };
+
+  selectedGoodHandler = (good: string) => {
+    this.setState(
+      {
+        nameGood: good,
+        selectedGood: `${good} is selected`,
+      },
+    );
+  };
+
+  clearHandler = () => {
+    this.setState(
+      {
+        nameGood: '',
+        selectedGood: 'No goods selected',
+      },
+    );
+  };
+
+  render() {
+    const { nameGood, selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          Selected good:
+          {selectedGood}
+          {' '}
+          <button
+            type="button"
+            onClick={() => {
+              this.clearHandler();
+            }}
+            className={classNames({ hide: nameGood === '' })}
+          >
+            X
+          </button>
+        </h1>
+        <ul>
+          {goodsFromServer.map(good => (
+            <li
+              className="list-item"
+              key={good}
+            >
```

Comment:
```suggestion
            <li className="list-item" key={good}>
```
Два атрибута можно оставлять так, если атрибутов больше - можно переносить